proposals/lip-0053.md

```diff
@@ -482,13 +552,25 @@ def execute(ccu: Transaction) -> None:
         # This function never raises an error.
         elif ccm.receivingChainID != CHAIN_ID_MAINCHAIN:
             forward(ccu, ccm)
-
         # We append at the very end. This implies that if the message leads to a chain termination,
         # it is still possible to recover it (because the channel terminated message
         # would refer to an inbox where the message has not been appended yet).
         appendToInboxTree(ccu.params.sendingChainID, ccmBytes)
     # Update the context to indicate that now we stop the CCM processing.
     ctx.ccmProcessing = False
+
+    # Update sidechain validators.
+    validatorsUpdate = ccu.params.activeValidatorsUpdate
+    if len(validatorsUpdate.blsKeysUpdate) > 0 or len(validatorsUpdate.bftWeightsUpdate) > 0 or validatorsUpdate.bftWeightsUpdateBitmap != EMPTY_BYTES or ccu.params.certificateThreshold != validators(ccu.params.sendingChainID).certificateThreshold:
+        updateValidators(ccu)
+
+    # Update last certificate.
+    if len(ccu.params.certificate) > 0:
+        updateCertificate(ccu)
+
+    # Update partner chain outbox root.
+    if ccu.params.inboxUpdate is not empty: # An empty object has all properties set to their default values.
```

Comment:
Minor comment: I think it would be good to change this condition to 

```suggestion
    if ccu.params.inboxUpdate is not empty and len(ccu.params.certificate) > 0:
```

The reason being that if certificate is empty, then the partner chain outbox root has been already updated earlier, so no need to re-calculate it. Same in line 713 for the sidechain CCU. I don't say the current is wrong, but adding this extra check avoids un-necessary computation and as far as I see it does not break any case.

Reply:
done